solver/src/main/java/org/chocosolver/util/objects/Weightable.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2025, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.util.objects;
+
+import gnu.trove.map.hash.TIntDoubleHashMap;
+import org.chocosolver.solver.Identity;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * <br/>
+ *
+ * @author Charles Prud'homme
+ * @since 26/03/2025
+ */
+public interface Weightable<I extends Identity> {
```

Comment:
I understand the interest of allowing different implementations to identify the best one. However, now we know the array is faster, shouldn't we keep only the best one ?